### PR TITLE
Add media module initial implementiation

### PR DESCRIPTION
**Pull request by joehoyle**
- Tachyon-plugin
- Smart media
- Rekognition
- Gaussholder
- Documentation

### CODE REVIEW

File: inc/namespace.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Bootstrap function to set up the module.
+ *
+ * Called from the HM Platform module loader if the module is activated.
+ *
+ * @return void
+ */
+function bootstrap() {
+	add_action( 'plugins_loaded', function () {
```
Comment (rmccue) [inline inc/namespace.php]: Can we break all of these anonymous functions out to full functions?

File: inc/namespace.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		}
+
+		if ( $config['smart-media'] ) {
+			require_once $vendor_dir . '/humanmade/smart-media/plugin.php';
+			remove_filter( 'intermediate_image_sizes_advanced', 'HM\\Media\\Cropper\\prevent_thumbnail_generation' );
+		}
+
+		if ( $config['gaussholder'] ) {
+			if ( ! empty( $config['gaussholder']['image-sizes'] ) ) {
+				add_filter( 'gaussholder.image_sizes', function ( $sizes ) use ( $config ) {
```
Comment (rmccue) [inline inc/namespace.php]: Break this out to a separate function and call `get_config()` again, IMO

File: composer.json [hunk 3]
```diff
@@ -8,5 +8,16 @@
 			"name": "Human Made",
 			"email": "hello@humanmade.com"
 		}
-	]
+	],
+	"require": {
+		"humanmade/tachyon-plugin": "^0.9.2",
+		"humanmade/smart-media": "^0.1.11",
+		"humanmade/gaussholder": "dev-master",
```
Comment (rmccue) [inline composer.json]: We should get this switched to a proper version ASAP.

File: docs/lazy-loading.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Configuration
+
+Lazy loading of images in post content is enabled by default. This behavior can be disabled by setting the `modules.media.gaussholder` property to `false` in your project's `composer.json`.
+
+Lazy loading is enabled on a per image size basis, so you must configure the specific image sizes. This is done via the `image-sizes` key in the configuration:
+
+```json
+"extra": {
+	"platform": {
```
Comment (rmccue) [inline docs/lazy-loading.md]: Indentation in this block is off.

File: inc/namespace.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * Called from the HM Platform module loader if the module is activated.
+ *
+ * @return void
+ */
+function bootstrap() {
+	add_action( 'plugins_loaded', __NAMESPACE__ . '\\load_plugins', 1 );
+}
+
+function load_plugins() {
```
Comment (rmccue) [inline inc/namespace.php]: Missing phpDoc.

File: docs/lazy-loading.md (hunk 4 above)
Comment (joehoyle) [reply docs/lazy-loading.md]: Hmm where?

Comment (rmccue) [reply docs/lazy-loading.md]: Aha, GitHub rendering bug with tabs. Disregard!

**Pull request merged**